waku/incentivization/rpc_codec.nim

```diff
@@ -0,0 +1,81 @@
+import
+  std/options
+import
+  ../common/protobuf,
+  ../waku_core,
+  ./rpc
+
+
+proc encode*(rpc: DummyRequest): ProtoBuffer = 
+  var pb = initProtoBuffer()
+
+  pb.write3(1, rpc.requestId)
+
+  if rpc.eligibilityProof.isSome:
+    let ep = rpc.eligibilityProof.get()
+    pb.write3(2, uint32(ord(ep.proofType)))
+    pb.write3(3, ep.proof)
+  else:
+    pb.write3(2, uint32(ord(EligibilityProofType.NONE)))
```

Comment:
I'd decouple this encoding from the encapsulating Request encoding. In other words, there should be a separate `encode*(rpc: EligibilityProof): ProtoBuffer`. Same for the `EligibilityStatus` and for decoding.